modules/guides/pages/import.adoc

```diff
@@ -0,0 +1,335 @@
+= Import Data
+// Import and Stream Data
+:page-topic-type: guide
+:imagesdir: ../assets/images
+:tabs:
+:description: How to import documents into Couchbase server.
+
+[abstract]
+{description}
+
+== Introduction
+
+Importing data can be done from the Couchbase Server UI,
+via the `cbimport` command-line tool shipped with Couchbase Server,
+or using the SDK to script the process.
+
+XXX server/current/tools/cbimport.html
+
+Data load essentially consists of:
+
+* Prepare data in some well known format such as Comma Separated Values or a stream of JSON documents
+
+* Connect to the Couchbase instance
+
+* Iterate over every document in the data load
+
+  * Decide on the key for this document (could be an ID, a sequence number, or some combination of fields)
+  
+  * Do any additional processing
+
+  * Upsert the document
+
+
+== Preparing the Data
+
+Extract or generate your data in the appropriate data format. The following are well supported for export as well as by `cbimport` and the module ecosystems of all our SDKs.
+
+
+[{tabs}]
+====
+CSV::
++
+--
+Comma Separated Values (.csv) are easily exported from many spreadsheet and database applications.
+
+Typically, the first row is a header row containing the names of the columns within the document.
+
+[source,csv]
+----
+include::nodejs-sdk:howtos:example$import.csv[]
+----
+
+{github}
+--
+
+TSV::
++
+--
+Tab Separated Values (.csv) are a common variant of CSV files.
+
+[source,tsv]
+----
+include::nodejs-sdk:howtos:example$import.tsv[]
+----
+
+{github}
+--
+
+JSON::
++
+--
+JSON (.json) files are especially well suited to import into Couchbase, as it is the default native datatype. As we want to import a number of objects, and a .json file contains only one single value, we will typically provide a .json with an *array* of the values to import.
```

Comment:
This is a good example of the point I was trying to make above (L46). We don't know whether the user wants to import a number of objects, although it's a good guess that they do.

If you decide that this section is instructional (how to lay out a file for import), you could reword like this:

```suggestion
JSON (.json) files are especially well suited to import into Couchbase, as it is the default native datatype.
A JSON file can only contain one single value.
If you want to import a number of objects, create a JSON file with an *array* of the values to import.
```

Reply:
Alternatively, if you decide that this section is descriptive (of the layout of a typical JSON file), you could reword like this:

```suggestion
JSON (.json) files are especially well suited to import into Couchbase, as it is the default native datatype.
A JSON file can contains only one single value.
If you need to import a number of objects,  the JSON file will typically contain an *array* of the values to import.
```

I'll leave you to decide which approach to take for this particular section, consistently of course 😀 , but for the most part I think you should default to the instructional approach for the how-to guides.

Reply:
Though giving an `if` suggests that users may later ask for an `else`... I can say 

> To let you import one or any number of objects as needed, create an array...

Also, it's possible (though unlikely) that they may want to store arrays, strings, etc.
Which is fine (but the "process by adding an import key" example later would then break)

Um, just thinking out loud, I think, will play around and submit again when ready for review!